src/devices/src/virtio/net/persist.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Defines the structures needed for saving/restoring net devices.
+
+use std::sync::atomic::AtomicUsize;
+use std::sync::Arc;
+
+use rate_limiter::{persist::RateLimiterState, RateLimiter};
+use snapshot::Persist;
+use versionize::{VersionMap, Versionize, VersionizeResult};
+use versionize_derive::Versionize;
+use vm_memory::GuestMemoryMmap;
+
+use super::*;
+
+use crate::virtio::persist::VirtioDeviceState;
+use crate::virtio::{DeviceState, Queue};
+
+#[derive(Versionize)]
+pub struct NetConfigSpace {
+    guest_mac: Vec<u8>,
+}
```

Comment:
@ioanachirca @sandreim what do you think about moving `config_space` to generic `VirtioDeviceState`.
The `config_space: Vec<u8>` is part of the generic virtio device spec.

Independent devices implementations would then interpret that generic `Vec<u8>` as they need it.

Reply:
While this is more generic, vsock doesn't have a `config_space` and the block's `config_space` contains the disk size (and if the disk changes size between save and restore, overwriting the config with the old size would maybe lead to errors). I think these are 2 cases out of 3 in which we wouldn't use the config on restore.

Reply:
That's right, but each device can do any impl-specific processing to the restored generic `config_space` pre-save/post-restore.

I'm just worried that we're ignoring saving parts of the spec based on assumptions stemming from our current device implementations.

Reply:
Makes sense. I added a commit that introduces `config_space` in `VirtioDeviceState` and only for the net device overwrites the config space on restore.

Reply:
I just spoke to @sandreim and we have to keep it as a defined struct to be able to versionize it. So let's keep it like you proposed it in this PR.
Marking as resolved.